### DISCUSSION TITLE
Remove Items from Stencil

### DISCUSSION THREAD

**Discussion by RajeshNakkanaboina** [clientIO/joint#2247] [Q&A]
### Introduction

Hi Team, 

I want to how can we give multi select checkbox to Items in Stencil, As my use case like to remove multiple selected items. 
Also want to how can we remove an item from stencil dynamically (on external button click or action)
Thanks In advance

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
How to add and remove elements from the _stencil_ is shown in the following demos:
https://www.jointjs.com/demos/dynamic-stencil
https://www.jointjs.com/demos/finite-object-pool

Note that the _template_ view is just a set of _papers_. Therefore, you can render basically anything that can be rendered in a regular `dia.Paper`.
https://www.jointjs.com/demos/tree-stencil

If you want to add HTML checkboxes next to each shape, it might be easier to render the whole _stencil_ in HTML as shown here:
https://codepen.io/jointjs/pen/vYVvjpW

**Reply by RajeshNakkanaboina** [reply]
Hi Kumilingus,
Thanks for the reply

![Capture](https://github.com/clientIO/joint/assets/134581112/b91f59c0-5cfc-40f4-974c-67b40b3c5064)
It is necessary to add checkboxes to a stencil like the one in the previous screenshot by clicking an external button and to remove the same by clicking a button. What techniques do I need to employ for this, if you could help me out?

**Reply by kumilingus** [reply]
I'd use a custom highlighter (you can decide whether to show the checkboxes or not at any point) and the `<foreignObject/>` with actual HTML `<input/>` (make it accessible by default).

Here's an [example](https://codepen.io/jointjs/pen/abQyLew/3b7585af60dea24c8fceb7e65d282d04?editors=1010).
